src/web/services/deadline-extension-helper.ts

```diff
@@ -1,3 +1,14 @@
+import {
+  InstructorExtensionTableColumnModel,
+  StudentExtensionTableColumnModel,
+} from '../app/pages-instructor/instructor-session-individual-extension-page/extension-table-column-model';
+import { Instructor, Student } from '../types/api-output';
+
+export enum DeadlineHandlerType {
+  CREATE,
+  DELETE,
+}
```

Comment:
Would it be a lot of work to remove this enum altogether?

This seems like a duplicate of `ExtensionModalType`, but more importantly the utility class shouldn't really need to care about what type of modal is calling its functions (which is basically all this enum does). This creates unnecessary coupling between the two classes.

This way, we can split up `getUpdatedDeadlines` into create and delete, or remove this method entirely. In general, passing in an argument that significantly changes the behavior of a method is usually thought of as a [code smell](https://martinfowler.com/bliki/FlagArgument.html).

Reply:
Done, please take a look and see if the refactoring was ok. Originally it was done this way to reduce the code duplication, but yea it increased the coupling.